### PR TITLE
⚡️ Enhance locationiq - formattedAddress, county, unify country name

### PR DESCRIPTION
**Pull request by a-tokyo**
⚡️ Enhance locationiq geocoder:
- add formattedAddress
- add county
- unify country name

closes https://github.com/nchaulet/node-geocoder/issues/336
closes https://github.com/nchaulet/node-geocoder/issues/337

### CODE REVIEW

**Comment by nchaulet** [comment]
For the unify country name it is still an issue? Did you try to contact locationIQ so they can fix the problem on their side? I am not a fan of hardcoding data here, (if it's the only solution maybe but I would like us to try to see if the provider can fix their data before)

**Comment by A-Tokyo** [comment]
@nchaulet Hey! Thanks for the response ( : 

I am not a fan of hardcoding either, however, unfortunately requesting locationIQ to change their API with a breaking change is not feasible. 

If you think I should remove the country name unification part, let me know. I already hardcoded the fix in my backend so this wouldn't affect my needs.

I think the other changes in this PR are crucial tho!

File: lib/geocoder/locationiqgeocoder.js [hunk 1]
```diff
@@ -120,25 +120,31 @@ LocationIQGeocoder.prototype._reverse = function(query, callback) {
...
+    /* make sure country name is unified accross geocoders */
+    transformedResult.country = result.address.country === 'United States of America' ? 'United States' : result.address.country;
     transformedResult.city = result.address.city || result.address.town || result.address.village || result.address.hamlet;
     transformedResult.state = result.address.state;
     transformedResult.zipcode = result.address.postcode;
     transformedResult.streetName = result.address.road || result.address.cycleway;
     transformedResult.streetNumber = result.address.house_number;
+    if (transformedResult.county) {
+      transformedResult.county = result.address.county;
+    }
```
Comment (A-Tokyo) [inline lib/geocoder/locationiqgeocoder.js:141]:
```suggestion
```

**Comment by nchaulet** [comment]
Hi @A-Tokyo Yes I think it probably make sense to remove the country specificity here, otherwise great job here, as soon you removed the hardcoded country it should be good to go

File: lib/geocoder/locationiqgeocoder.js [hunk 2]
```diff
@@ -120,25 +120,32 @@ LocationIQGeocoder.prototype._reverse = function(query, callback) {
...
 
+  if (result.display_name) {
+    transformedResult.formattedAddress = result.display_name;
+  }
+
   if (result.address) {
-    transformedResult.country = result.address.country;
-    transformedResult.country = result.address.country;
+    /* make sure country name is unified accross geocoders */
+    transformedResult.country = result.address.country === 'United States of America' ? 'United States' : result.address.country;
```
Comment (A-Tokyo) [inline lib/geocoder/locationiqgeocoder.js:133]:
```suggestion
    transformedResult.country = result.address.country;
```

**Comment by A-Tokyo** [comment]
@nchaulet Done! 👍🏻

**Pull request merged**